sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Implicits.scala

```diff
@@ -48,23 +48,27 @@ private[sql] object CatalogV2Implicits {
   }
 
   implicit class TransformHelper(transforms: Seq[Transform]) {
-    def asPartitionColumns: Seq[String] = {
+    def validatePartitionColumns(): Unit = {
       val (idTransforms, nonIdTransforms) = transforms.partition(_.isInstanceOf[IdentityTransform])
 
       if (nonIdTransforms.nonEmpty) {
         throw new AnalysisException("Transforms cannot be converted to partition columns: " +
-            nonIdTransforms.map(_.describe).mkString(", "))
+          nonIdTransforms.map(_.describe).mkString(", "))
```

Comment:
@rdblue do we allow to use bucket transform as a partition column? It's not allowed in `ResolveSessionCatalog.scala`, but there is a test in `DataFrameWriterV2Suite.scala` testing `test("Create: partitioned by bucket(4, id)")`.

Reply:
That table property allows is used to make the test table implementation accept configuration that it doesn't support when writing. It's used to test that the table was passed the right `Transform`, even though the `InMemoryTable` only supports identity transforms.

ResolveSessionCatalog should convert bucket Transforms to and from BucketSpec.